File: SwiftTestsForObjCTests/ViewControllerTests.m
//
//  ViewControllerTests.m
//  SwiftTestsForObjC
//
//  Created by Andy Obusek on 3/23/17.
//
//

#import <XCTest/XCTest.h>
#import "ViewController.h"

// Private Category Trick to expose private properties for testing
@interface ViewController (Test)

@property (strong, nonatomic) UILabel *nameLabel;
@property (strong, nonatomic) UILabel *emailLabel;

@end

@interface ViewControllerTests : XCTestCase

@end

@implementation ViewControllerTests

- (void)testViewDidLoad_SetsLabels_Always {
  NSString *name = @"Andy";
  NSString *email = @"andyo@notanemail.com";
  [[NSUserDefaults standardUserDefaults] setObject:name forKey:@"name"];
  [[NSUserDefaults standardUserDefaults] setObject:email forKey:@"email"];

  ViewController *toTest = [[ViewController alloc] init];
  [toTest viewDidLoad];
  XCTAssertEqualObjects(name, toTest.nameLabel.text);
  XCTAssertEqualObjects(email, toTest.emailLabel.text);
}

@end

File: SwiftTestsForObjCTests/ViewControllerTests.swift
//
//  ViewControllerTests.swift
//  SwiftTestsForObjC
//
//  Created by Andy Obusek on 3/23/17.
//
//

import XCTest
@testable import SwiftTestsForObjC

class TestViewController: ViewController {
  dynamic var nameLabel: UILabel!
  dynamic var emailLabel: UILabel!
}

class ViewControllerTests: XCTestCase {

  func testViewDidLoad_SetsLabels_Always() {
    let name = "Andy"
    let email = "andy@notanemail.com"
    UserDefaults.standard.set(name, forKey: "name")
    UserDefaults.standard.set(email, forKey: "email")

    let toTest = TestViewController()
    toTest.viewDidLoad()

    XCTAssertEqual(name, toTest.nameLabel.text)
    XCTAssertEqual(email, toTest.emailLabel.text)
  }
    
}
